File: eureka/ios/nativeLib/nativeLib/Models/LoggerStruct.swift
//
//  LoggerStruct.swift
//  BLEDemo
//

import Foundation

public class LoggerStruct {
  
  private final var _log: [String : String] = [:]
  
  public init(_ pFunctionName:String, pFileName:String, pEventDescription:String) {
    setValues(pFunctionName: pFunctionName, pFileName: pFileName, pErrorCode: nil, pEventDescription: pEventDescription, pLineNumber: "");
  }

  public init(_ pFunctionName: String, pFileName:String, pErrorCode:ResultCodeEnum, pEventDescription: String, pLineNumber: String) {
      setValues(pFunctionName: pFunctionName, pFileName: pFileName, pErrorCode: pErrorCode, pEventDescription: pEventDescription, pLineNumber: pLineNumber)
  }
  
  public func setValues(pFunctionName: String, pFileName: String, pErrorCode: ResultCodeEnum?, pEventDescription: String, pLineNumber:String) {
    
    let userID = Global.getUserId() <= 0 ? Global.getUserIdForScan() : Global.getUserId()
    let watchMSN = Global.getWatchMSN().count <= 0 ? Global.getWatchMSNForScan() : Global.getWatchMSN()
    
    _log["ts"] = "\(NSDate().timeIntervalSince1970 * 1000)"
    _log["userId"] = userID.description
    _log["seq_no"] = "\(LpLogger.getNewSrlNo())"
    _log["userName"] = ""
    _log["codeCategoryId"] = "1"
    _log["functionName"] = pFunctionName
    _log["fileName"] = pFileName
    _log["operationName"] = "add"
    _log["watchMSN"] = watchMSN
    _log["osVersion"] = ""
    _log["errorCode"] = (pErrorCode == nil) ? "" : pErrorCode?.code
    _log["eventDescription"] = pEventDescription
    _log["lineNumber"] = pLineNumber
    _log["applicationNameId"] = "1"
  }
  
  internal func getLog() -> [String:String] {
    return _log
  }
  
  public func getJSONStr() -> String {
    let encoder = JSONEncoder()
    if let jsonData = try? encoder.encode(_log) {
      if let jsonString = String(data: jsonData, encoding: .utf8) {
        return jsonString
      }
    }
    return ""
  }
}

File: eureka/android/library/src/main/java/com/lifeplus/Pojo/LoggerStruct.java
package com.lifeplus.Pojo;

import android.os.Build;

import com.google.gson.Gson;
import com.lifeplus.Pojo.Enum.ResultCodeEnum;
import com.lifeplus.Util.Global;
import com.lifeplus.Util.LpLogger;

import java.util.Date;
import java.util.HashMap;

public class LoggerStruct {
    private final HashMap<String, String> mLog = new HashMap<>();

    public LoggerStruct(String pFunctionName, String pFileName, String pEventDescription) {
        setValues(pFunctionName, pFileName, null, pEventDescription, "");
    }

    public LoggerStruct(String pFunctionName, String pFileName,
                        ResultCodeEnum pErrorCode, String pEventDescription, String pLineNumber) {
        setValues(pFunctionName, pFileName, pErrorCode, pEventDescription, pLineNumber);
    }

    private void setValues(String pFunctionName, String pFileName,
                           ResultCodeEnum pErrorCode, String pEventDescription, String pLineNumber) {
        mLog.put("ts", String.valueOf(new Date().getTime()));
        mLog.put("userId", String.valueOf((Global.getUserId() <= 0) ? Global.getUserIdForScan() : Global.getUserId()));
        mLog.put("seq_no", String.valueOf(LpLogger.getNewSrlNo()));
        mLog.put("userName", "");
        mLog.put("codeCategoryId", "2");
        mLog.put("functionName", pFunctionName);
        mLog.put("fileName", pFileName);
        mLog.put("operationName", "add");
        mLog.put("watchMSN", (Global.getWatchMSN().length() <= 0) ? Global.getWatchMSNForScan() : Global.getWatchMSN());
        mLog.put("appVersion", Global.getAppVersion());
        mLog.put("osVersion", Build.VERSION.RELEASE);
        mLog.put("errorCode", (pErrorCode == null) ? "" : pErrorCode.getCode());
        mLog.put("eventDescription", pEventDescription);
        mLog.put("lineNumber", pLineNumber);
        mLog.put("applicationNameId", "1");
    }

    public String getJSONStr() {
        return new Gson().toJson( mLog );
    }
}
